### PR TITLE
Replace operator-sdk e2e tests with testenv

### PR DESCRIPTION
**Pull request by qinqon**
**Is this a BUG FIX or a FEATURE ?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug
> /kind enhancement

**What this PR does / why we need it**:
This allow use to remove the operator-sdk CLI dependency and to remove
the dependency on deprecated operator-sdk e2e api.

**Special notes for your reviewer**:
Depends on https://github.com/nmstate/kubernetes-nmstate/pull/610

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/env/env.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+	nmstatev1alpha1 "github.com/nmstate/kubernetes-nmstate/api/v1alpha1"
+	nmstatev1beta1 "github.com/nmstate/kubernetes-nmstate/api/v1beta1"
+	"github.com/nmstate/kubernetes-nmstate/test/environment"
+)
+
+var (
+	cfg               *rest.Config
+	Client            client.Client         // You'll be using this client in your tests.
+	KubeClient        *kubernetes.Clientset // You'll be using this client in your tests.
```
Comment (rhrazdil) [inline test/env/env.go:20]: I can only see the Client (line above) to be used. What is the difference?

File: test/e2e/handler/main_test.go [hunk 2]
```diff
@@ -14,18 +14,17 @@ import (
 	ginkgoreporters "kubevirt.io/qe-tools/pkg/ginkgo-reporters"
 
 	corev1 "k8s.io/api/core/v1"
-	dynclient "sigs.k8s.io/controller-runtime/pkg/client"
 
-	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	//knmstatereporter "github.com/nmstate/kubernetes-nmstate/test/reporter"
```
[the remark this answers is not in the document]
Comment (qinqon) [reply test/e2e/handler/main_test.go:18]: I forgot about this one we have to re-activate it.

File: test/env/env.go (hunk 1 above)
Comment (qinqon) [reply test/env/env.go:20]: First one is controller-runtime client it has different API and a cache, second one is client-go one does not have a cache, let me check if we are really not using it.

Comment (qinqon) [reply test/env/env.go:20]: It's used by the reporter 
```golang
func writePodsLogs(writer io.Writer, namespace string, sinceTime time.Time) {   
    podLogOpts := corev1.PodLogOptions{}                                        
    podLogOpts.SinceTime = &metav1.Time{Time: sinceTime}                        
    podList := &corev1.PodList{}                                                
    err := testenv.Client.List(context.TODO(), podList, &dynclient.ListOptions{})
    Expect(err).ToNot(HaveOccurred())                                           
    podsClientset := testenv.KubeClient.CoreV1().Pods(namespace)                
                                                                                
    for _, pod := range podList.Items {                                         
```

File: test/e2e/handler/main_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (qinqon) [reply test/e2e/handler/main_test.go:18]: Done

File: test/e2e/handler/main_test.go [hunk 3]
```diff
@@ -39,39 +38,37 @@ var (
...
+	resetDesiredStateForNodes()
+})
+
+func TestE2E(t *testing.T) {
+
+	RegisterFailHandler(Fail)
+
 	reporters := make([]Reporter, 0)
-	reporters = append(reporters, knmstatereporter.New("test_logs/e2e/handler", framework.Global.OperatorNamespace, nodes))
+	//reporters = append(reporters, knmstatereporter.New("test_logs/e2e/handler", testenv.OperatorNamespace, nodes))
```
[the remark this answers is not in the document]
Comment (qinqon) [reply test/e2e/handler/main_test.go:71]: Done

**Comment by qinqon** [comment]
[force-push](https://github.com/nmstate/kubernetes-nmstate/compare/0e18991193f1cd16c30f5490b8c0ec6d94f9ba56..2516276bc1854ce89c90f887d67f2125fea124b2): addressed @rhrazdil comments

**Comment by qinqon** [comment]
Since we want to remove operator-sdk cli to bump to golang 1.15 I am going to leave reporter as it was at handler and we fix it later on, ok with it @rhrazdil ?

**Review by bcrochet — APPROVED** [review]

**Pull request merged**